### ISSUE TITLE
Recycled view holders retain old auto-sized size.

### ISSUE THREAD

**Issue by JakeWharton** [JakeWharton/SdkSearch#36]
Check out `Toolbar.SavedState` compared to its neighbors above and below. Framework bug?

![screenshot_20180116-151727](https://user-images.githubusercontent.com/66577/35010324-84a18f72-fad0-11e7-9820-fd755f439b34.png)

cc @chrisbanes

**Comment by chrisbanes** [comment]
Yeah, looks like a framework bug to me. Either that or ConstraintLayout is doing something weird, but I doubt it's that.

**Comment by chrisbanes** [comment]
Worked it out. The issue is that ConstraintLayout for some reason is caching the TextView's height. This then trips up TextView's internal logic for whether it needs to do a full layout (which it does if the size changes).

**Comment by JakeWharton** [comment]
This can be easily reproduced by searching for "View" and then doing moderate flings down to the bottom and then back up. Usually the 4th or 5th item from the top will retain the smaller size from the recycled views.

**Comment by JakeWharton** [comment]
Got a 100% repro case. On a Nexus 4 emulator, search "Toolbar", rotate left, scroll to bottom, rotate right, scroll to top.

![screen shot 2018-05-31 at 11 42 47 pm](https://user-images.githubusercontent.com/66577/40820066-56dca4ca-652c-11e8-8cd3-dd316d5b3ea2.png)
